src/components/RepositoryDetail/RepositoryDetail.tsx

```diff
@@ -0,0 +1,372 @@
+import React from 'react'
+import Link from 'next/link'
+import { Button } from 'react-bootstrap';
+import { gql } from '@apollo/client'
+import {FACET_FIELDS, Facet} from '../FacetList/FacetList'
+import VerticalBarChart from '../VerticalBarChart/VerticalBarChart'
+import DonutChart, { typesRange, typesDomain } from '../DonutChart/DonutChart'
+import ProductionChart from '../ProductionChart/ProductionChart'
+import {
+  EmailShareButton,
+  FacebookShareButton,
+  TwitterShareButton
+} from 'react-share'
+import { faEnvelope, faNewspaper } from '@fortawesome/free-solid-svg-icons'
+import { faTwitter, faFacebook } from '@fortawesome/free-brands-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faSignInAlt } from '@fortawesome/free-solid-svg-icons'
+
+import { compactNumbers } from '../../utils/helpers'
+import {
+  RepositoriesNode,
+  REPOSITORY_FIELDS,
+}from '../RepositoryMetadata/RepositoryMetadata'
+import styles from './RepositoryDetail.module.scss'
+
+export const REPOSITORY_DETAIL_FIELDS = gql`
+  ${REPOSITORY_FIELDS}
+  ${FACET_FIELDS}
+  fragment repositoryDetailFields on Repository{
+    ...repoFields
+    citationCount
+    downloadCount
+    viewCount
+    works {
+      totalCount
+      languages{...facetFields}
+      resourceTypes{...facetFields}
+      fieldsOfScience{...facetFields}
+      authors{...facetFields}
+      licenses{...facetFields}
+      published{...facetFields}
+    }
+  }
+`
+export interface RepositoryWorks {
+  totalCount: number
+  languages: [Facet]
+  resourceTypes: [Facet]
+  fieldsOfScience: [Facet]
+  authors: [Facet]
+  licenses: [Facet]
+  published: [Facet]
+}
+
+export interface RepositoryDetailNode extends RepositoriesNode{
+  citationCount: number
+  downloadCount: number
+  viewCount: number
+  works: RepositoryWorks
+  contacts: [string]
+}
+
+type Props = {
+  repo: RepositoryDetailNode
+}
+
+function facetToData(facetList){
+  return facetList.map((x) => ({
+    title: x.title,
+    count: x.count
+  }))
+}
+
+
+type ConditionaBarChartProps = {
+  title: string,
+  data: [Facet]
+}
+type ConditionaDonutChartProps = {
+  title: string,
+  count: number,
+  data: [Facet]
+}
+
+type EmptyChartProps ={
+  title: string
+}
+
+const EmptyChart: React.FunctionComponent<EmptyChartProps> = ({title}) => {
+  return (
+    <div className={`panel panel-transparent`}>
+    <div className={`panel-body ${styles.emptyChart}`}>
+      <div className={styles.title}>
+        <h4>{title}</h4>
+      </div>
+      <div>Not enough data to render</div>
+  </div>
+  </div>
+)
+}
+
+const ConditionalProductionChart: React.FunctionComponent<ConditionaBarChartProps> = ({title, data}) => {
```

Comment:
Just a comment, I was wondering whether this would be better suited to be a feature of the charts components rather than a Component that wraps the chart component?

Reply:
I think you are right.
Expect an  update

Reply:
Refactored